app/components/shared/structured_rejection_reasons_component.rb

```diff
@@ -0,0 +1,44 @@
+class StructuredRejectionReasonsComponent < ViewComponent::Base
+  include ViewHelper
+
+  attr_reader :application_choice, :editable
+
+  def initialize(application_choice:, editable: false, render_link_to_find_when_rejected_on_qualifications: false)
+    @application_choice = application_choice
+    @editable = editable
+    @render_link_to_find_when_rejected_on_qualifications = render_link_to_find_when_rejected_on_qualifications
+  end
+
+  def reasons
+    structured_rejection_reasons_class.new(application_choice.structured_rejection_reasons)
+  end
+
+  def component
+    component_class.new(
+      application_choice: application_choice,
+      reasons: reasons,
+      editable: editable,
+      render_link_to_find_when_rejected_on_qualifications: @render_link_to_find_when_rejected_on_qualifications,
+    )
+  end
+
+  def component_class
+    if @application_choice.rejection_reasons_type == 'rejection_reasons'
+      RejectionReasons::RejectionReasonsComponent
+    else
+      RejectionReasons::ReasonsForRejectionComponent
+    end
+  end
+
+  def structured_rejection_reasons_class
+    if @application_choice.rejection_reasons_type == 'rejection_reasons'
+      RejectionReasons
+    else
```

Comment:
if we default to the new rejection reason this means we will render it regardless of type being there or not, as we've lost the check for `present?` now.

Reply:
I've reworked the logic with how we get the presenter or component class.
The fallback is to use the rejection reason (single simple text) version, as this isn't opinionated about the data.
I've also added/reinstated escape clauses for when there's no rejection reasons data so we don't end up with empty component summary rows (eg. https://github.com/DFE-Digital/apply-for-teacher-training/pull/6723/files#diff-15a8fd49292ffe71c92eb4d1f093a07ac30301aea11b1607f92b900a76d76980R174)